### PR TITLE
Support HTTP HEAD, OPTIONS, and PUT

### PR DESCRIPTION
**Pull request by ospencer**

### CODE REVIEW

File: rwasm/api/api_http.go [hunk 1]
```diff
@@ -11,10 +11,13 @@ import (
...
-	methodPost   = int32(2)
-	methodPatch  = int32(3)
-	methodDelete = int32(4)
+	methodGet     = int32(0)
+	methodHead    = int32(1)
+	methodOptions = int32(2)
+	methodPost    = int32(3)
+	methodPut     = int32(4)
+	methodPatch   = int32(5)
+	methodDelete  = int32(6)
```
Comment (jagger27) [inline rwasm/api/api_http.go:20]:
```suggestion
const (
	methodGet int32 = iota
	methodHead
	methodOptions
	methodPost
	methodPut
	methodPatch
	methodDelete
)

Exact same thing, just with the fancy Go `iota` idiom. 
```

File: api/swift/Sources/suborbital/lib.swift [hunk 2]
```diff
@@ -44,19 +44,34 @@ public func Set(runnable: Runnable) {
     RUNNABLE = runnable
 }
 
-let httpMethodGet = Int32(1)
-let httpMethodPost = Int32(2)
-let httpMethodPatch = Int32(3)
-let httpMethodDelete = Int32(4)
+let httpMethodGet = Int32(0)
```
Comment (jagger27) [inline api/swift/Sources/suborbital/lib.swift:47]:
```swift
enum HTTPMethod: Int32 {
    case GET
    case HEAD
    case OPTIONS
    case POST
    case PUT
    case PATCH
    case DELETE
}

public func HttpGet(url: String) -> String {
    return fetch(method: .GET, url: url, body: "")
}

public func HttpHead(url: String) -> String {
    return fetch(method: .HEAD, url: url, body: "")
}

public func HttpOptions(url: String) -> String {
    return fetch(method: .OPTIONS, url: url, body: "")
}

public func HttpPost(url: String, body: String) -> String {
    return fetch(method: .POST, url: url, body: body)
}

public func HttpPut(url: String, body: String) -> String {
    return fetch(method: .PUT, url: url, body: body)
}

public func HttpPatch(url: String, body: String) -> String {
    return fetch(method: .PATCH, url: url, body: body)
}

public func HttpDelete(url: String) -> String {
    return fetch(method: .DELETE, url: url, body: "")
}

func fetch(method: HTTPMethod, url: String, body: String) -> String {
```
This sparks a little more joy.

Comment (ospencer) [reply api/swift/Sources/suborbital/lib.swift:47]: @willemneal That's just Swift! You have to opt out of having a named parameter when you define the function if you don't want the caller to have to provide it.

Comment (willemneal) [reply api/swift/Sources/suborbital/lib.swift:47]: Ah I see.  And it seems to also have default parameters so you could remove some of the `body: ""`s.

**Review by cohix — APPROVED** [review]

**Pull request merged**